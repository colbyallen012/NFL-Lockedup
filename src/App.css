.main { 
  display: grid; 
  grid-template-columns: repeat(5, 20vw); 
  grid-template-rows: repeat(5, 20vh); 
  grid-column-gap: 0px;
  grid-row-gap: 0px; 
 }

  .header { 
    grid-area: 1 / 2 / 1 / 6; 
    display: flex;
    flex-direction: column;
    justify-content: center;
  } 

  h1 {
    text-align: center;
  }

  .crimes { 
    grid-area: 2 / 2 / 2 / 6; 
  } 

  .teams { 
    background-color: #D50A0A;
    grid-area: 1 / 1 / 8 / 2;
  }

  .team-name {
    text-align: center;
  }

  .arrests { 
    grid-area: 4 / 2 / 6 / 6; 
  }