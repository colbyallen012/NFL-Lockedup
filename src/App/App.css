.main { 
  display: grid; 
  grid-template-columns: repeat(5, 20vw); 
  grid-template-rows: repeat(5, 20vh); 
  grid-column-gap: 0px;
  grid-row-gap: 0px; 
 }

  .header { 
    display: flex;
    flex-direction: row;
    justify-content: center;
    background-color: white;
    grid-area: 1 / 2 / 1 / 6; 
  } 

  .logo {
    height: 120px;
    margin: auto 10px;
    width: 120px;
  }

  .title {
    color: #013369;
    font-family: Graduate;
    font-size: 90px;
    text-align: center;
    text-shadow: 3px 3px #D50A0A; 
    margin: auto 10px;
  }

  .crimes { 
    display: flex;
    flex-direction: row;
    justify-content: center;
    grid-area: 2 / 2 / 2 / 6; 
    margin: 10px 0px;
  } 

  .teams { 
    background-color: #D50A0A;
    grid-area: 1 / 1 / 8 / 2;
  }

  .team-header {
    background-color: white;
    border-radius: 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    margin: 0px auto;
    width: 92%;
  }

  .team-name {
    color: #013369;
    font-family: Graduate;
    font-size: 30px;
    letter-spacing: 1px;
    margin: auto 15px;
    text-align: center;
    text-shadow: 1px 1px #D50A0A;
  }

  .team-logo {
    height: 80px;
    margin: auto 15px;
    width: 90px;
  }

  .arrests { 
    grid-area: 4 / 2 / 6 / 6; 
    margin: 10px 0px;
  }