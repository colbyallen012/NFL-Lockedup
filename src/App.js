import React, {Component} from 'react';
import { getCrimesData, getPlayerData, getPositionData } from './apiCalls/apiCalls'
import { connect } from 'react-redux';
import { showCrimes, showPlayers, showPositions, showTeamArrests } from './Actions';
import {Route} from 'react-router-dom'
import CrimesContainer from './Containers/CrimesContainer/CrimesContainer'
import TeamsContainer from './Containers/TeamsContainer/TeamsContainer'
import TeamNavBar from './Containers/TeamNavBar/TeamNavBar'
import Home from './Components/Home/Home'
import './App.css';

export class App extends Component {

  async componentDidMount() {
    getCrimesData()
      .then(crimes => this.props.showCrimes(crimes))

    getPlayerData()
      .then(players => this.props.showPlayers(players))

    getPositionData()
      .then(positions => this.props.showPositions(positions))
  }

  render () {
    return (
        <main className='main'>
          <header className='header'>
            <h1>NFL LockedUp</h1>
          </header>
          <section className='crimes'>
            <CrimesContainer crimes={this.props.crimes} players={this.props.players} positions={this.props.positions}/>
          </section>
          <section className='teams'>
            <TeamNavBar crimes={this.props.crimes} players={this.props.players} />
          </section>
          <section className='arrests'>
            <Route exact path='/' component={Home}/>
            <Route exact path='/Den' render={() =>
              <div>
                <h2 className='team-name'>Broncos</h2>
              <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Min' render={() => 
              <div>
                <h2 className='team-name'>Vikings</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Cin' render={() => 
              <div>
                <h2 className='team-name'>Bengals</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Jac' render={() => 
              <div>
                <h2 className='team-name'>Jaguars</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Tb' render={() => 
              <div>
                <h2 className='team-name'>Bucaneers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Ten' render={() => 
              <div>
                <h2 className='team-name'>Titans</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Cle' render={() => 
              <div>
                <h2 className='team-name'>Browns</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Ind' render={() => 
              <div>
                <h2 className='team-name'>Colts</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Kc' render={() => 
              <div>
                <h2 className='team-name'>Chiefs</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Chi' render={() => 
              <div>
                <h2 className='team-name'>Bears</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Mia' render={() => 
              <div>
                <h2 className='team-name'>Dolphins</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Sea' render={() => 
              <div>
                <h2 className='team-name'>Seahawks</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Sf' render={() => 
              <div>
                <h2 className='team-name'>49ers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Bal' render={() => 
              <div>
                <h2 className='team-name'>Ravens</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Lac' render={() => 
              <div>
                <h2 className='team-name'>Chargers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/No' render={() => 
              <div>
                <h2 className='team-name'>Saints</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Gb' render={() => 
              <div>
                <h2 className='team-name'>Packers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Pit' render={() => 
              <div>
                <h2 className='team-name'>Steelers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Nyj' render={() => 
              <div>
                <h2 className='team-name'>Jets</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Ari' render={() => 
              <div>
                <h2 className='team-name'>Cardinals</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Oak' render={() => 
              <div>
                <h2 className='team-name'>Raiders</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Was' render={() => 
              <div>
                <h2 className='team-name'>Redskins</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Dal' render={() => 
              <div>
                <h2 className='team-name'>Cowboys</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Car' render={() => 
              <div>
                <h2 className='team-name'>Panthers</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Buf' render={() => 
              <div>
                <h2 className='team-name'>Bills</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Ne' render={() => 
              <div>
                <h2 className='team-name'>Patriots</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Atl' render={() => 
              <div>
               <h2 className='team-name'>Falcons</h2>
               <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Phi' render={() => 
              <div>
               <h2 className='team-name'>Eagles</h2>
               <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/Nyg' render={() => 
              <div>
                <h2 className='team-name'>Giants</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
            <Route exact path='/La' render={() => 
              <div>
                <h2 className='team-name'>Rams</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
             </div>
            }/>
            <Route exact path='/Det' render={() => 
              <div>
                <h2 className='team-name'>Lions</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>
             <Route exact path='/Hou' render={() => 
              <div>
                <h2 className='team-name'>Texans</h2>
                <TeamsContainer arrests={this.props.teamArrests}/> 
              </div>
            }/>             
          </section>
        </main>
    )
  }
}

const mapStateToProps = (store) => ({
  crimes: store.crimes,
  players: store.players,
  positions: store.positions,
  teamArrests: store.teamArrests
})

const mapDispatchToProps = dispatch => ({
  showCrimes: (crimes) => dispatch(showCrimes(crimes)),
  showPlayers: (players) => dispatch(showPlayers(players)),
  showPositions: (positions) => dispatch(showPositions(positions)),
  showTeamArrests: (team) => dispatch(showTeamArrests(team))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);

