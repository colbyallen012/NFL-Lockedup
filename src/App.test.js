import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  let instance;

  beforeEach(()=> {
    wrapper = shallow(<App />);
    instance = wrapper.instance();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

});