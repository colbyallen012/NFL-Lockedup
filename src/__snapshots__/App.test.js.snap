// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="main"
>
  <header
    className="header"
  >
    <h1>
      NFL LockedUp
    </h1>
  </header>
  <section
    className="crimes"
  >
    <CrimesContainer />
  </section>
  <section
    className="teams"
  >
    <Connect(TeamNavBar) />
  </section>
  <section
    className="arrests"
  >
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/Den"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Min"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Cin"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Jac"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Tb"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Ten"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Cle"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Ind"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Kc"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Chi"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Mia"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Sea"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Sf"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Bal"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Lac"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/No"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Gb"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Pit"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Nyj"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Ari"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Oak"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Was"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Dal"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Car"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Buf"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Ne"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Atl"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Phi"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Nyg"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/La"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Det"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/Hou"
      render={[Function]}
    />
  </section>
</main>
`;
